server-spi-private/src/main/java/org/keycloak/logging/MappedDiagnosticContextProvider.java

```diff
@@ -0,0 +1,71 @@
+package org.keycloak.logging;
+
+import org.keycloak.models.ClientModel;
+import org.keycloak.models.KeycloakContext;
+import org.keycloak.models.OrganizationModel;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserSessionModel;
+import org.keycloak.provider.Provider;
+import org.keycloak.sessions.AuthenticationSessionModel;
+
+/**
+ * Provider interface for updating the Mapped Diagnostic Context (MDC) with key/value pairs based on the current keycloak context.
+ * All keys in the MDC will be prefixed with "kc." to avoid conflicts.
+ *
+ * @author <a href="mailto:b.eicki@gmx.net">Björn Eickvonder</a>
+ */
+public interface MappedDiagnosticContextProvider extends Provider {
+
+    static final String MDC_PREFIX = "kc.";
```

Comment:
`static final` is redundant in interfaces. 
```suggestion
    String MDC_PREFIX = "kc.";
```